### ISSUE TITLE
Problem with Syntax error on token "RestrictedIdentifierWhen", Identifier expected?

### ISSUE THREAD

**Issue by Hellblazer** [eclipse-jdt/eclipse.jdt.core#498]
Eclipse:
Version: 2022-12 M1 (4.26.0 M1)
Build id: 20221006-1952

I am using Java 19 and I do _not_ have "--enable-preview" as a default jam argument.

The mockito method: Mockito.when() now creates the syntax error.

It seems that this should not occur by default as JEP 427 is only a third preview and requires "--enable-preview" to enable.

My code is part of a maven build, and I _do_ have "--enable-preview" enabled for the maven compiler, and when compiling this code with Java 19 I do not see this syntax error nor any warnings.  So I'm not sure what the issue is.

Is there some problem with my config/etc that would resolve this?  I cannot seem to find any solution to this, unfortunately.

The change seems to be [here](https://github.com/eclipse-jdt/eclipse.jdt.core/blob/ecd24c483047682ca28164ff8c2db365b2826a47/org.eclipse.jdt.core/grammar/java.g#L233).

<img width="993" alt="image" src="https://user-images.githubusercontent.com/4781/198349323-d821127c-33d8-49c2-b134-8924ff53cd73.png">

**Comment by Hellblazer** [comment]
Huh.  Okay, so I tried to create a project to isolate this problem with a simple example.  Not reproducible.  I've tried a clean project, importing my maven project with the issue into Eclipse anew.  The problem reproduces.  Very odd.

Note this problem appeared with the new eclipse version, reverting does not reproduce.  If I use JDK 18, does not reproduce.

So, closing this as it's obviously something extremely weird with this maven project (although completely unobvious).  Will update when I figure it out.
